srcs/python/kungfu/torch/ops/collective.py

```diff
@@ -41,3 +43,9 @@ def broadcast_parameters(state_dict):
         h = inplace_broadcast_async_op(value, name)
         handles.append(h)
     wait_all_handles(handles)
+
+def all_gather(x):
+    np = current_cluster_size()
+    y = torch.zeros((x.size(0) * np, x.size(1)))
```

Comment:
are you assuming x is a rank 2 tensor here?

Reply:
For a tensor with shape (d1, d2, ..., dr), the result of all_gather should have shape (np, d1, ..., dr).